graphql-cli/src/main/java/io/ballerina/graphql/generator/CodeGenerator.java

```diff
@@ -83,8 +83,9 @@ public void generate(GraphqlProject project) throws GenerationException {
      * @throws TypesGenerationException             when a types code generation error occurs
      * @throws IOException                          If an I/O error occurs
      */
-    private List<SrcFilePojo> generateBalSources(GraphqlProject project) throws ClientGenerationException,
-            UtilsGenerationException, TypesGenerationException, IOException, ConfigTypesGernerationException {
+    public List<SrcFilePojo> generateBalSources(GraphqlProject project, GeneratedContext generatedContext)
```

Comment:
IMO we can rename the context as `GeneratorContext` instead. WDYT?
```suggestion
    public List<SrcFilePojo> generateBalSources(GraphqlProject project, GeneratorContext generatorContext)
```